src/transformer/mockDefiner/mockDefiner.ts

```diff
@@ -250,10 +270,12 @@ export class MockDefiner {
     if (this._factoryRegistrationsPerFile[sourceFile.fileName]) {
       return this._factoryRegistrationsPerFile[sourceFile.fileName]
         .map((reg: { key: ts.Declaration; factory: ts.Expression }) => {
-          const key: string = this._factoryCache.get(reg.key);
+          const key: string | undefined = this._factoryCache.get(reg.key);
```

Comment:
This cannot fail. We should improve the types

Reply:
Disabling `@typecript-eslint/no-non-null-assertion` would allow us to replace this (and the other places) with the following if we can guarantee that the key will be available at all times.

```
const key: string = this._factoryCache.get(reg.key)!;
```